static/scripts/pay.ts

```diff
@@ -38,12 +58,21 @@ export const pay = async (): Promise<void> => {
   claimButtonElem.addEventListener("click", async () => {
     try {
       const signer = await connectWallet();
-      await withdraw(signer, txData);
+      const balance = await fetchTreasury();
+      await toggleStatus(balance);
+      if (balance >= Number(txData.permit.permitted.amount)) {
+        await withdraw(signer, txData);
+      } else {
+        alert("Not enough funds on treasury to claim");
```

Comment:
Don't use alerts. You should dump the output like I did in the DOM

Reply:
okay